.github/workflows/main_push_workflow.yml

```
@@ -22,4 +22,4 @@ jobs:
           java-version: ${{ matrix.java-version }}
           distribution: temurin
       - name: Build with Gradle
-        run: ./gradlew build releaseTarGz
+        run: ./gradlew build integrationTest releaseTarGz
```

Wouldn't `check` be enough?
```suggestion
        run: ./gradlew check releaseTarGz
```

---

check does not run integration tests unfortunately

---

You're right! I always get confused by the docs https://docs.gradle.org/current/userguide/java_testing.html#sec:configuring_java_integration_tests but it adds it explicitly to the check with `dependsOn`. I guess we don't see a need to add it to the check lifecycle

---

Also talking about this specific place I don't think releaseTarGz is needed here since we are not publishing the result anywhere

---

I think it was added to exercise the packaging as it's used on the Docker image (maybe we should build the image as part of the workflow instead?) cc @ivanyu

---

We perhaps should, changed it back to build + releaseTarGz since build runs all the tests now. Will figure out releaseTarGz necessity and image building later.

---

sounds good to me